ui/src/app/shared/components/yaml/yaml-editor.tsx

```diff
@@ -41,7 +42,7 @@ export class YamlEditor<T> extends React.Component<Props<T>, State> {
                         autoFocus={true}
                     />
                 ) : (
-                    <YamlViewer value={this.state.value} />
+                    <YamlViewer value={jsYaml.dump(this.props.value)} />
```

Comment:
Why is this change necessary?

Reply:
Consider the following use case:
* A cronworkflow is running
* A user clicks `suspend` and the cronworkflow is suspended
* The user then edits the workflow and submits the updated version

Without this change, the update will fail since the suspended workflow is not reflected in the editor, i.e., `suspend: true` is missing, and an error will be returned asking the user to submit with a newer version of the cronworkflow.

Therefore, we need to use the prop value to display the updated value.

Reply:
Correct, but if `this.props.value` is changed, the entire react component would re-render. I.e., the constructor would get called again and `this.state.value` would get filled with the new prop value. This is a feature of react.

Do you still think we need this change?

Reply:
Initially I had the same thought as yours, but it turned out that the editor didn't rerender.

Reply:
In this case can we use this solution: https://stackoverflow.com/a/51309488? I'd like us to stay consistent in using `this.state.value` across the file.

Reply:
Updated